units/commit_queue.go

```diff
@@ -265,80 +266,99 @@ func (j *commitQueueJob) processGitHubPRItem(ctx context.Context, cq *commitqueu
+				pr.Base.User.GetLogin(),
+				pr.Base.Repo.GetName(),
+				pr.Head.GetRef(),
+				InvalidConfig,
+			)
+			update.Run(ctx)
+			j.AddError(update.Error())
+			j.logError(catcher.Resolve(), "invalid config file", nextItem)
+			j.AddError(sendCommitQueueGithubStatus(j.env, pr, message.GithubStateFailure, "can't make patch", ""))
+			j.dequeue(cq, nextItem)
+			return
+		}
 
-	if err = writePatchInfo(patchDoc, patchSummaries, patch); err != nil {
-		j.logError(err, "can't make patch", nextItem)
-		j.AddError(sendCommitQueueGithubStatus(j.env, pr, message.GithubStateFailure, "can't make patch", ""))
-		j.dequeue(cq, nextItem)
-		return
-	}
+		if err = writePatchInfo(patchDoc, patchSummaries, patch); err != nil {
+			j.logError(err, "can't make patch", nextItem)
+			j.AddError(sendCommitQueueGithubStatus(j.env, pr, message.GithubStateFailure, "can't make patch", ""))
+			j.dequeue(cq, nextItem)
+			return
+		}
 
-	modulePRs, modulePatches, dequeue, err := getModules(ctx, githubToken, nextItem, projectConfig)
-	if err != nil {
-		j.logError(err, "can't get modules", nextItem)
-		if dequeue {
-			j.AddError(sendCommitQueueGithubStatus(j.env, pr, message.GithubStateFailure, "can't get modules", ""))
+		modulePRs, modulePatches, dequeue, err := getModules(ctx, githubToken, nextItem, projectConfig)
+		if err != nil {
+			j.logError(err, "can't get modules", nextItem)
+			if dequeue {
+				j.AddError(sendCommitQueueGithubStatus(j.env, pr, message.GithubStateFailure, "can't get modules", ""))
+				j.dequeue(cq, nextItem)
+			} else {
+				j.logError(cq.SetProcessing(false), "can't set processing to false", nextItem)
+			}
+			return
+		}
+		patchDoc.Patches = append(patchDoc.Patches, modulePatches...)
+
+		// populate tasks/variants matching the commitqueue alias
+		projectConfig.BuildProjectTVPairs(patchDoc, patchDoc.Alias)
+
+		if err = addMergeTaskAndVariant(patchDoc, projectConfig, projectRef, commitqueue.SourcePullRequest); err != nil {
+			j.logError(err, "can't set patch project config", nextItem)
+			event.LogCommitQueueEnqueueFailed(nextItem.Issue, err)
 			j.dequeue(cq, nextItem)
-		} else {
-			j.logError(cq.SetProcessing(false), "can't set processing to false", nextItem)
+			return
 		}
-		return
-	}
-	patchDoc.Patches = append(patchDoc.Patches, modulePatches...)
 
-	// populate tasks/variants matching the commitqueue alias
-	projectConfig.BuildProjectTVPairs(patchDoc, patchDoc.Alias)
+		if err = patchDoc.Insert(); err != nil {
+			j.logError(err, "can't insert patch", nextItem)
+			j.dequeue(cq, nextItem)
+			j.AddError(sendCommitQueueGithubStatus(j.env, pr, message.GithubStateFailure, "can't make patch", ""))
+			return
+		}
 
-	if err = addMergeTaskAndVariant(patchDoc, projectConfig, projectRef, commitqueue.SourcePullRequest); err != nil {
-		j.logError(err, "can't set patch project config", nextItem)
-		event.LogCommitQueueEnqueueFailed(nextItem.Issue, err)
-		j.dequeue(cq, nextItem)
-		return
+		for _, modulePR := range modulePRs {
```

Comment:
Why do we only want to send modulePR statuses in this case?

Reply:
it's more that module PRs don't get their status updated elsewhere, so this is as good a place as any when we know the main PR had a patch successfully created

Reply:
What I mean is, this is in the if statement that you intend to delete. I see that you moved the "send status to module PRs" to AddPatchForPr in rest/data which would cover the new case, but this means that the module status will show up when the patch is created, not when the patch is finalized; is that an issue?

Reply:
Gotcha. At basically the same time, we send a status of pending with description "added to queue" for the main patch. I changed the description to be the same for the modules. I added a todo when I clean things up to also do this when we finalize, so that I don't have to handle the backwards compatibility scenario for this one since it'll be ugly